pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -63,80 +65,170 @@ def __init__(
+            )
+        )
+
+    def build_model(
+        self,
+        data: Optional[Union[np.ndarray, pd.DataFrame, pd.Series]] = None,
+        model_config: Optional[Dict] = None,
+        **kwargs,
+    ) -> None:
+        self.output_var = "target"
+        self.data = self.generate_model_data(data=data)
 
-        with pm.Model(coords=coords) as self.model:
+        if not model_config:
+            model_config = self.default_model_config
+        with pm.Model(coords=self.model_coords) as self.model:
             channel_data_ = pm.MutableData(
                 name="channel_data",
-                value=channel_data,
-                dims=("date", "channel"),
+                value=self.data.channel_data_.value,
+                dims=self.data.channel_data_.dims,
             )
 
-            target_ = pm.MutableData(name="target", value=target_data, dims="date")
+            target_ = pm.MutableData(
+                name="target",
+                value=self.data["target_data_"]["value"],
+                dims=self.data["target_data_"]["dims"],
+            )
 
-            intercept = pm.Normal(name="intercept", mu=0, sigma=2)
+            intercept = pm.Normal(
+                name="intercept",
+                mu=model_config["intercept"]["mu"],
+                sigma=model_config["intercept"]["sigma"],
+            )
 
+            """beta_channel = pm.HalfNormal(
+                name="beta_channel",
+                sigma=model_config["beta_channel"]["sigma"],
+                dims=model_config["beta_channel"]["dims"],
+            )  # ? Allow prior depend on channel costs?"""
             channel_prior = self._preprocess_channel_prior()
             beta_channel = self.model.register_rv(
                 rv_var=channel_prior, name="beta_channel", dims="channel"
             )
 
-            alpha = pm.Beta(name="alpha", alpha=1, beta=3, dims="channel")
+            alpha = pm.Beta(
+                name="alpha",
+                alpha=model_config["alpha"]["alpha"],
+                beta=model_config["alpha"]["beta"],
+                dims=model_config["alpha"]["dims"],
+            )
 
-            lam = pm.Gamma(name="lam", alpha=3, beta=1, dims="channel")
+            lam = pm.Gamma(
+                name="lam",
+                alpha=model_config["lam"]["alpha"],
+                beta=model_config["lam"]["beta"],
+                dims=model_config["lam"]["dims"],
+            )
 
-            sigma = pm.HalfNormal(name="sigma", sigma=2)
+            sigma = pm.HalfNormal(name="sigma", sigma=model_config["sigma"]["sigma"])
 
             channel_adstock = pm.Deterministic(
                 name="channel_adstock",
                 var=geometric_adstock(
                     x=channel_data_,
                     alpha=alpha,
-                    l_max=adstock_max_lag,
+                    l_max=kwargs["adstock_max_lag"]
+                    if "adstock_max_lag" in kwargs
+                    else 4,
```

Comment:
I would not set a default as I think users should know with `adstock_max_lag` they need.

Reply:
as above, this was just my attempt of creating the sandbox for new pymc users, but if it's something you're decided on not having I can make it as required parameter, and remove the default values